cornice_swagger/views.py

```diff
@@ -0,0 +1,73 @@
+import importlib
+import pkg_resources
+from string import Template
+
+import cornice
+import cornice_swagger
+from pyramid.response import Response
+
+
+def swagger_ui_template_view(request):
+    """
+    Serves Swagger UI page, default Swagger UI config is used but you can
+    override the callable that generates the `<script>` tag by setting
+    `cornice_swagger.swagger_ui_script_generator` in pyramid config, it defaults
+    to 'cornice_swagger.views:swagger_ui_script_template'
+
+    :param request:
+    :return:
+    """
+    script_generator = request.registry.settings.get(
+        'cornice_swagger.swagger_ui_script_generator',
+        'cornice_swagger.views:swagger_ui_script_template')
+    package, callable = script_generator.split(':')
+    imported_package = importlib.import_module(package)
+    script_callable = getattr(imported_package, callable)
+    template = pkg_resources.resource_string(
+        'cornice_swagger', 'templates/index.html').decode('utf8')
+
+    # hardcode for now since that will work for vast majority of users
+    # maybe later add minified resources for behind firewall support?
+    ui_css_url = 'https://cdnjs.cloudflare.com/ajax/libs/swagger-ui/3.12.9/swagger-ui.css'
+    ui_js_bundle_url = 'https://cdnjs.cloudflare.com/ajax/libs/' \
+                       'swagger-ui/3.12.9/swagger-ui-bundle.js'
+    ui_js_standalone_url = 'https://cdnjs.cloudflare.com/ajax/libs/' \
+                           'swagger-ui/3.12.9/swagger-ui-standalone-preset.js'
+    html = Template(template).safe_substitute(
+        ui_css_url=ui_css_url,
+        ui_js_bundle_url=ui_js_bundle_url,
+        ui_js_standalone_url=ui_js_standalone_url,
+        swagger_ui_script=script_callable(request),
+    )
+    return Response(html)
+
+
+def swagger_json_view(request):
+    """
+    :param request:
+    :return:
+
+    Generates JSON representation of Swagger spec
+    """
+    doc = cornice_swagger.CorniceSwagger(cornice.service.get_services())
+    kwargs = request.registry.settings['cornice_swagger.spec_kwargs']
+    my_spec = doc.generate(**kwargs)
+    return my_spec
```

Comment:
I think there should be an example on how to replace this with a custom view. 

For example, in Kinto we inherit the `CorniceSwagger` class and instantiate our own stuff the the view :) 

https://github.com/Kinto/kinto/blob/8db533939f47d33c44deb66e0d1285258d99d68b/kinto/core/views/openapi.py#L22-L32

Reply:
I think if you want to replace this with a custom view - then you don't use the directive, because it basicly just adds 2 views so you don't have to override them. If someone wants to override they'd have to redo all the work directive does basicly.